File: binding/include/libqplugin.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpluginmetadata.html
pub const qpluginmetadata = struct {

    pub fn ArchRequirements() u8 {
        return qtc.QPluginMetaData_ArchRequirements();
    }

    pub fn Size(self: ?*anyopaque, ) u64 {
        return qtc.QPluginMetaData_Size(@ptrCast(self));
    }

    pub fn SetSize(self: ?*anyopaque, size: u64) void {
        qtc.QPluginMetaData_SetSize(@ptrCast(self), size);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPluginMetaData_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qstaticplugin.html
pub const qstaticplugin = struct {

    /// New constructs a new QStaticPlugin object.
    pub fn New(other: ?*anyopaque) QtC.QStaticPlugin {
        return qtc.QStaticPlugin_new(@ptrCast(other));
    }


    /// New2 constructs a new QStaticPlugin object.
    pub fn New2(other: ?*anyopaque) QtC.QStaticPlugin {
        return qtc.QStaticPlugin_new2(@ptrCast(other));
    }


    /// New3 constructs a new QStaticPlugin object.
    pub fn New3(param1: ?*anyopaque) QtC.QStaticPlugin {
        return qtc.QStaticPlugin_new3(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QStaticPlugin_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QStaticPlugin_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn MetaData(self: ?*anyopaque, ) QtC.QJsonObject {
        return qtc.QStaticPlugin_MetaData(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QStaticPlugin_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpluginmetadata-header.html
pub const qpluginmetadata__header = struct {

    /// New constructs a new QPluginMetaData::Header object.
    pub fn New(other: ?*anyopaque) QtC.QPluginMetaData__Header {
        return qtc.QPluginMetaData__Header_new(@ptrCast(other));
    }


    /// New2 constructs a new QPluginMetaData::Header object.
    pub fn New2(other: ?*anyopaque) QtC.QPluginMetaData__Header {
        return qtc.QPluginMetaData__Header_new2(@ptrCast(other));
    }


    /// New3 constructs a new QPluginMetaData::Header object.
    pub fn New3(param1: ?*anyopaque) QtC.QPluginMetaData__Header {
        return qtc.QPluginMetaData__Header_new3(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__Header_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__Header_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Version(self: ?*anyopaque, ) u8 {
        return qtc.QPluginMetaData__Header_Version(@ptrCast(self));
    }

    pub fn SetVersion(self: ?*anyopaque, version: u8) void {
        qtc.QPluginMetaData__Header_SetVersion(@ptrCast(self), version);
    }

    pub fn MajorVersion(self: ?*anyopaque, ) u8 {
        return qtc.QPluginMetaData__Header_MajorVersion(@ptrCast(self));
    }

    pub fn SetQtMajorVersion(self: ?*anyopaque, qt_major_version: u8) void {
        qtc.QPluginMetaData__Header_SetQtMajorVersion(@ptrCast(self), qt_major_version);
    }

    pub fn MinorVersion(self: ?*anyopaque, ) u8 {
        return qtc.QPluginMetaData__Header_MinorVersion(@ptrCast(self));
    }

    pub fn SetQtMinorVersion(self: ?*anyopaque, qt_minor_version: u8) void {
        qtc.QPluginMetaData__Header_SetQtMinorVersion(@ptrCast(self), qt_minor_version);
    }

    pub fn PluginArchRequirements(self: ?*anyopaque, ) u8 {
        return qtc.QPluginMetaData__Header_PluginArchRequirements(@ptrCast(self));
    }

    pub fn SetPluginArchRequirements(self: ?*anyopaque, plugin_arch_requirements: u8) void {
        qtc.QPluginMetaData__Header_SetPluginArchRequirements(@ptrCast(self), plugin_arch_requirements);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPluginMetaData__Header_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpluginmetadata-magicheader.html
pub const qpluginmetadata__magicheader = struct {

    /// New constructs a new QPluginMetaData::MagicHeader object.
    pub fn New(other: ?*anyopaque) QtC.QPluginMetaData__MagicHeader {
        return qtc.QPluginMetaData__MagicHeader_new(@ptrCast(other));
    }


    /// New2 constructs a new QPluginMetaData::MagicHeader object.
    pub fn New2(other: ?*anyopaque) QtC.QPluginMetaData__MagicHeader {
        return qtc.QPluginMetaData__MagicHeader_new2(@ptrCast(other));
    }


    /// New3 constructs a new QPluginMetaData::MagicHeader object.
    pub fn New3() QtC.QPluginMetaData__MagicHeader {
        return qtc.QPluginMetaData__MagicHeader_new3();
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__MagicHeader_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__MagicHeader_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn Header(self: ?*anyopaque, ) QtC.QPluginMetaData__Header {
        return qtc.QPluginMetaData__MagicHeader_Header(@ptrCast(self));
    }

    pub fn SetHeader(self: ?*anyopaque, header: QtC.QPluginMetaData__Header) void {
        qtc.QPluginMetaData__MagicHeader_SetHeader(@ptrCast(self), @ptrCast(header));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPluginMetaData__MagicHeader_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/qpluginmetadata-elfnoteheader.html
pub const qpluginmetadata__elfnoteheader = struct {

    /// New constructs a new QPluginMetaData::ElfNoteHeader object.
    pub fn New(other: ?*anyopaque) QtC.QPluginMetaData__ElfNoteHeader {
        return qtc.QPluginMetaData__ElfNoteHeader_new(@ptrCast(other));
    }


    /// New2 constructs a new QPluginMetaData::ElfNoteHeader object.
    pub fn New2(other: ?*anyopaque) QtC.QPluginMetaData__ElfNoteHeader {
        return qtc.QPluginMetaData__ElfNoteHeader_new2(@ptrCast(other));
    }


    /// New3 constructs a new QPluginMetaData::ElfNoteHeader object.
    pub fn New3(payloadSize: u32) QtC.QPluginMetaData__ElfNoteHeader {
        return qtc.QPluginMetaData__ElfNoteHeader_new3(payloadSize);
    }


    /// New4 constructs a new QPluginMetaData::ElfNoteHeader object.
    pub fn New4(param1: ?*anyopaque) QtC.QPluginMetaData__ElfNoteHeader {
        return qtc.QPluginMetaData__ElfNoteHeader_new4(@ptrCast(param1));
    }

    /// CopyAssign shallow copies `other` into `self`.
    pub fn CopyAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__ElfNoteHeader_CopyAssign(@ptrCast(self), @ptrCast(other));
    }

    /// MoveAssign moves `other` into `self` and invalidates `other`.
    pub fn MoveAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        qtc.QPluginMetaData__ElfNoteHeader_MoveAssign(@ptrCast(self), @ptrCast(other));
    }


    pub fn NNamesz(self: ?*anyopaque, ) u32 {
        return qtc.QPluginMetaData__ElfNoteHeader_NNamesz(@ptrCast(self));
    }

    pub fn SetNNamesz(self: ?*anyopaque, n_namesz: u32) void {
        qtc.QPluginMetaData__ElfNoteHeader_SetNNamesz(@ptrCast(self), n_namesz);
    }

    pub fn NDescsz(self: ?*anyopaque, ) u32 {
        return qtc.QPluginMetaData__ElfNoteHeader_NDescsz(@ptrCast(self));
    }

    pub fn SetNDescsz(self: ?*anyopaque, n_descsz: u32) void {
        qtc.QPluginMetaData__ElfNoteHeader_SetNDescsz(@ptrCast(self), n_descsz);
    }

    pub fn NType(self: ?*anyopaque, ) u32 {
        return qtc.QPluginMetaData__ElfNoteHeader_NType(@ptrCast(self));
    }

    pub fn SetNType(self: ?*anyopaque, n_type: u32) void {
        qtc.QPluginMetaData__ElfNoteHeader_SetNType(@ptrCast(self), n_type);
    }

    pub fn Header(self: ?*anyopaque, ) QtC.QPluginMetaData__Header {
        return qtc.QPluginMetaData__ElfNoteHeader_Header(@ptrCast(self));
    }

    pub fn SetHeader(self: ?*anyopaque, header: QtC.QPluginMetaData__Header) void {
        qtc.QPluginMetaData__ElfNoteHeader_SetHeader(@ptrCast(self), @ptrCast(header));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPluginMetaData__ElfNoteHeader_Delete(@ptrCast(self));
    }
};

File: binding/src/libqplugin.cpp
#include <QJsonObject>
#include <QPluginMetaData>
#define WORKAROUND_INNER_CLASS_DEFINITION_QPluginMetaData__ElfNoteHeader
#define WORKAROUND_INNER_CLASS_DEFINITION_QPluginMetaData__Header
#define WORKAROUND_INNER_CLASS_DEFINITION_QPluginMetaData__MagicHeader
#include <QStaticPlugin>
#include <qplugin.h>
#include "libqplugin.h"
#include "libqplugin.hxx"

unsigned char QPluginMetaData_ArchRequirements() {
	return QPluginMetaData::archRequirements();
}

unsigned long long QPluginMetaData_Size(const QPluginMetaData* self) {
	return self->size;
}

void QPluginMetaData_SetSize(QPluginMetaData* self, unsigned long long size) {
	self->size;
}

void QPluginMetaData_Delete(QPluginMetaData* self) {
    delete self;
}

QStaticPlugin* QStaticPlugin_new(const QStaticPlugin* other) {
	 return new QStaticPlugin(*other);
}

QStaticPlugin* QStaticPlugin_new2(QStaticPlugin* other) {
	 return new QStaticPlugin(*other);
}

QStaticPlugin* QStaticPlugin_new3(const QStaticPlugin* param1) {
	 return new QStaticPlugin(*param1);
}

void QStaticPlugin_CopyAssign(QStaticPlugin* self, QStaticPlugin* other) {
    *self = *other;
}

void QStaticPlugin_MoveAssign(QStaticPlugin* self, QStaticPlugin* other) {
    *self = std::move(*other);
}

QJsonObject* QStaticPlugin_MetaData(const QStaticPlugin* self) {
	return new QJsonObject(self->metaData());
}

void QStaticPlugin_Delete(QStaticPlugin* self) {
    delete self;
}

QPluginMetaData__Header* QPluginMetaData__Header_new(const QPluginMetaData__Header* other) {
	 return new QPluginMetaData::Header(*other);
}

QPluginMetaData__Header* QPluginMetaData__Header_new2(QPluginMetaData__Header* other) {
	 return new QPluginMetaData::Header(*other);
}

QPluginMetaData__Header* QPluginMetaData__Header_new3(const QPluginMetaData__Header* param1) {
	 return new QPluginMetaData::Header(*param1);
}

void QPluginMetaData__Header_CopyAssign(QPluginMetaData__Header* self, QPluginMetaData__Header* other) {
    *self = *other;
}

void QPluginMetaData__Header_MoveAssign(QPluginMetaData__Header* self, QPluginMetaData__Header* other) {
    *self = std::move(*other);
}

unsigned char QPluginMetaData__Header_Version(const QPluginMetaData__Header* self) {
	return self->version;
}

void QPluginMetaData__Header_SetVersion(QPluginMetaData__Header* self, unsigned char version) {
	self->version;
}

unsigned char QPluginMetaData__Header_MajorVersion(const QPluginMetaData__Header* self) {
	return self->qt_major_version;
}

void QPluginMetaData__Header_SetQtMajorVersion(QPluginMetaData__Header* self, unsigned char qt_major_version) {
	self->qt_major_version;
}

unsigned char QPluginMetaData__Header_MinorVersion(const QPluginMetaData__Header* self) {
	return self->qt_minor_version;
}

void QPluginMetaData__Header_SetQtMinorVersion(QPluginMetaData__Header* self, unsigned char qt_minor_version) {
	self->qt_minor_version;
}

unsigned char QPluginMetaData__Header_PluginArchRequirements(const QPluginMetaData__Header* self) {
	return self->plugin_arch_requirements;
}

void QPluginMetaData__Header_SetPluginArchRequirements(QPluginMetaData__Header* self, unsigned char plugin_arch_requirements) {
	self->plugin_arch_requirements;
}

void QPluginMetaData__Header_Delete(QPluginMetaData__Header* self) {
    delete self;
}

QPluginMetaData__MagicHeader* QPluginMetaData__MagicHeader_new(const QPluginMetaData__MagicHeader* other) {
	 return new QPluginMetaData::MagicHeader(*other);
}

QPluginMetaData__MagicHeader* QPluginMetaData__MagicHeader_new2(QPluginMetaData__MagicHeader* other) {
	 return new QPluginMetaData::MagicHeader(*other);
}

QPluginMetaData__MagicHeader* QPluginMetaData__MagicHeader_new3() {
	 return new QPluginMetaData::MagicHeader();
}

void QPluginMetaData__MagicHeader_CopyAssign(QPluginMetaData__MagicHeader* self, QPluginMetaData__MagicHeader* other) {
    *self = *other;
}

void QPluginMetaData__MagicHeader_MoveAssign(QPluginMetaData__MagicHeader* self, QPluginMetaData__MagicHeader* other) {
    *self = std::move(*other);
}

QPluginMetaData__Header* QPluginMetaData__MagicHeader_Header(const QPluginMetaData__MagicHeader* self) {
	return new QPluginMetaData::Header(self->header);
}

void QPluginMetaData__MagicHeader_SetHeader(QPluginMetaData__MagicHeader* self, QPluginMetaData__Header* header) {
	self->header;
}

void QPluginMetaData__MagicHeader_Delete(QPluginMetaData__MagicHeader* self) {
    delete self;
}

QPluginMetaData__ElfNoteHeader* QPluginMetaData__ElfNoteHeader_new(const QPluginMetaData__ElfNoteHeader* other) {
	 return new QPluginMetaData::ElfNoteHeader(*other);
}

QPluginMetaData__ElfNoteHeader* QPluginMetaData__ElfNoteHeader_new2(QPluginMetaData__ElfNoteHeader* other) {
	 return new QPluginMetaData::ElfNoteHeader(*other);
}

QPluginMetaData__ElfNoteHeader* QPluginMetaData__ElfNoteHeader_new3(unsigned int payloadSize) {
	 return new QPluginMetaData::ElfNoteHeader(payloadSize);
}

QPluginMetaData__ElfNoteHeader* QPluginMetaData__ElfNoteHeader_new4(const QPluginMetaData__ElfNoteHeader* param1) {
	 return new QPluginMetaData::ElfNoteHeader(*param1);
}

void QPluginMetaData__ElfNoteHeader_CopyAssign(QPluginMetaData__ElfNoteHeader* self, QPluginMetaData__ElfNoteHeader* other) {
    *self = *other;
}

void QPluginMetaData__ElfNoteHeader_MoveAssign(QPluginMetaData__ElfNoteHeader* self, QPluginMetaData__ElfNoteHeader* other) {
    *self = std::move(*other);
}

unsigned int QPluginMetaData__ElfNoteHeader_NNamesz(const QPluginMetaData__ElfNoteHeader* self) {
	return self->n_namesz;
}

void QPluginMetaData__ElfNoteHeader_SetNNamesz(QPluginMetaData__ElfNoteHeader* self, unsigned int n_namesz) {
	self->n_namesz;
}

unsigned int QPluginMetaData__ElfNoteHeader_NDescsz(const QPluginMetaData__ElfNoteHeader* self) {
	return self->n_descsz;
}

void QPluginMetaData__ElfNoteHeader_SetNDescsz(QPluginMetaData__ElfNoteHeader* self, unsigned int n_descsz) {
	self->n_descsz;
}

unsigned int QPluginMetaData__ElfNoteHeader_NType(const QPluginMetaData__ElfNoteHeader* self) {
	return self->n_type;
}

void QPluginMetaData__ElfNoteHeader_SetNType(QPluginMetaData__ElfNoteHeader* self, unsigned int n_type) {
	self->n_type;
}

QPluginMetaData__Header* QPluginMetaData__ElfNoteHeader_Header(const QPluginMetaData__ElfNoteHeader* self) {
	return new QPluginMetaData::Header(self->header);
}

void QPluginMetaData__ElfNoteHeader_SetHeader(QPluginMetaData__ElfNoteHeader* self, QPluginMetaData__Header* header) {
	self->header;
}

void QPluginMetaData__ElfNoteHeader_Delete(QPluginMetaData__ElfNoteHeader* self) {
    delete self;
}

